src/test/java/org/broadinstitute/hellbender/utils/nio/LineIteratorTest.java

```diff
@@ -0,0 +1,52 @@
+package org.broadinstitute.hellbender.utils.nio;
+
+import com.google.common.jimfs.Configuration;
+import com.google.common.jimfs.Jimfs;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.FileSystem;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class LineIteratorTest {
+
+  @Test
+  public void testLineIteratorInForEach() throws IOException {
+    try (FileSystem jimfs = Jimfs.newFileSystem(Configuration.unix())) {
+      final Path path = jimfs.getPath("test.txt");
+      try (BufferedWriter bufferedWriter = Files.newBufferedWriter(path)) {
+        bufferedWriter.write("Hello world\n");
+        bufferedWriter.write("What's new?");
+      }
+      ArrayList<String> got = new ArrayList<>();
+      for (String s: new LineIterator(path)) {
```

Comment:
This doesn't close the `LineIterator` -- wrap the loop within a try-with-resources

Reply:
This isn't necessary, because the filesystem itself is in a try-with-resources so all its files will be closed.

But I changed the code anyways, so it shows the good examples. Normally, the filesystem is a long longer-lived and so it makes sense to close early.